loki/visitors/pretty_visualize.py

```diff
@@ -0,0 +1,349 @@
+
+        return [(node_info, edge_info)]
+
+    # Handler for outer objects
+    def visit_Module(self, o, **kwargs):
+        """
+        Add a :any:`Module`, mark parent node and visit all "spec" and "subroutine" nodes.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit(o.spec, **kwargs))
+        node_edge_info.extend(self.visit(o.subroutines, **kwargs))
+
+        return node_edge_info
+
+    def visit_Subroutine(self, o, **kwargs):
+        """
+        Add a :any:`Subroutine`, mark parent node and visit all "docstring", "spec", "body", "members" nodes.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit(o.docstring, **kwargs))
+        node_edge_info.extend(self.visit(o.spec, **kwargs))
+        node_edge_info.extend(self.visit(o.body, **kwargs))
+        node_edge_info.extend(self.visit(o.members, **kwargs))
+
+        return node_edge_info
+
+    # Handler for AST base nodes
+    def visit_Comment(self, o, **kwargs):
+        """
+        Enables turning off comments.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information, or list of nothing.
+        """
+        if kwargs.get("show_comments"):
+            return self.visit_Node(o, **kwargs)
+        return [None]
+
+    visit_CommentBlock = visit_Comment
+
+    def visit_Node(self, o, **kwargs):
+        """
+        Add a :any:`Node`, mark parent and visit all children.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit_all(o.children, **kwargs))
+        return node_edge_info
+
+    def visit_Expression(self, o, **kwargs):
+        """
+        Dispatch routine to add nodes utilizing expression tree stringifier,
+        mark parent and stop.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information or list of nothing.
+        """
+        if kwargs.get("show_expressions"):
+            content = self.symgen(o)
+            parent = kwargs.get("parent")
+            return self.__add_node(o, label=content, parent=parent, shape="box")
+        return [None]
```

Comment:
Does this need the `None` in the list? I'd imagine an empty list `[]` would have the same effect?